test/local/dispatcherTest.js

```diff
@@ -70,6 +71,7 @@ describe('Dispatcher Resolver', () => {
         };
 
         it('Basic products search', () => {
+            let searchProducts = sinon.spy(ProductsLoader.prototype, '__searchProducts');
```

Comment:
I would personally set the spies in a `beforeEach(() => {});` section and restore them in an `afterEach(() => {});` section. The way you use the `finally` for this only works if `resolve` doesn't throw an error.